stratum/hal/lib/nikss/nikss_wrapper.cc

```diff
@@ -38,25 +40,56 @@ ABSL_CONST_INIT absl::Mutex NikssWrapper::init_lock_(absl::kConstInit);
 
 NikssWrapper::NikssWrapper() {}
 
+::util::Status NikssWrapper::AddPort(int pipeline_id,
+                                     const std::string& port_name) {
+  auto ctx = absl::make_unique<nikss_context_t>();
+  nikss_context_init(ctx.get());
+  nikss_context_set_pipeline(ctx.get(), static_cast<nikss_pipeline_id_t>(pipeline_id));
+
+  int port_id = -1;
+  LOG(INFO) << "Adding port " << port_name << " to pipeline " << pipeline_id << ".";
+  RETURN_IF_NIKSS_ERROR(nikss_pipeline_add_port(ctx.get(), port_name.c_str(), &port_id));
+  LOG(INFO) << "Port added with port_id=" << port_id << ".";
+  nikss_context_free(ctx.get());
+  
+  return ::util::OkStatus();
+}
+
+::util::Status NikssWrapper::DelPort(int pipeline_id,
+                                     const std::string& port_name) {
+  auto ctx = absl::make_unique<nikss_context_t>();
+  nikss_context_init(ctx.get());
+  nikss_context_set_pipeline(ctx.get(), static_cast<nikss_pipeline_id_t>(pipeline_id));
+
+  RETURN_IF_NIKSS_ERROR(nikss_pipeline_del_port(ctx.get(), port_name.c_str()));
+
+  nikss_context_free(ctx.get());
+  
+  return ::util::OkStatus();
+}
+
 ::util::Status NikssWrapper::AddPipeline(int pipeline_id,
                                          const std::string bpf_obj) {
-  std::string tmp_filepath = "/tmp/stratum_bpf.o";
+  std::string tmp_filepath = "/etc/stratum/bpf.o";
   // FIXME: nikss currently doesn't support loading BPF programs from memory.
   //  So, we save it to the disk first and let NIKSS load it from the disk.
   RETURN_IF_ERROR(WriteStringToFile(bpf_obj, tmp_filepath));
 
   auto ctx = absl::make_unique<nikss_context_t>();
   nikss_context_init(ctx.get());
   nikss_context_set_pipeline(ctx.get(), static_cast<nikss_pipeline_id_t>(pipeline_id));
-
   if (nikss_pipeline_exists(ctx.get())) {
     LOG(INFO) << "NIKSS pipeline already exists, re-pushing is not supported yet.";
     return ::util::OkStatus();
   }
 
+  // FIXME: file is not removed if the load() fails
```

Comment:
Can we fix this issue in this PR?

Reply:
I think we can postpone this for later PR because it is not critical for now